mavros_extras/src/plugins/odom.cpp

```diff
@@ -142,52 +81,25 @@ class OdometryPlugin : public plugin::PluginBase {
-	std::string frame_id;				//!< parent frame identifier
-	std::string child_frame_id;			//!< child frame identifier
-
-	MAV_FRAME lf_id;				//!< local frame (pose) ID
-	MAV_FRAME bf_id;				//!< body frame (pose) ID
+	std::string fcu_odom_parent_id_des;			//!< desired orientation of the child frame (input data)
+	std::string fcu_odom_child_id_des;			//!< desired orientation of the body frame (input data)
+	std::string ext_odom_parent_id;				//!< parent frame of the ext odometry msg
+	std::string ext_odom_child_id;				//!< child frame of the ext odometry msg
 
 	/**
-	 * @brief Lookup transforms
-	 * @todo Implement in a more general fashion in the API IOT apply frame transforms
-	 * This should also run in parallel on a thread
-	 * @param[in] &frame_id The parent frame of reference
-	 * @param[in] &child_frame_id The child frame of reference
-	 * @param[in] &local_frame_orientation The desired local frame orientation
-	 * @param[in] &body_frame_orientation The desired body frame orientation
-	 * @param[in,out] &tf_parent2local The affine transform from the parent frame to the local frame
-	 * @param[in,out] &tf_child2local The affine transform from the child frame to the local frame
-	 * @param[in,out] &tf_parent2body The affine transform from the parent frame to the body frame
-	 * @param[in,out] &tf_child2body The affine transform from the child frame to the body frame
+	 * @brief Lookup transform with error handling
+	 * @param[in] &target The parent frame of the transformation you want to get
+	 * @param[in] &source The child frame of the transformation you want to get
+	 * @param[in,out] &tf_source2target The affine transform from the source to target
 	 */
-	void transform_lookup(const std::string &frame_id, const std::string &child_frame_id,
-		const std::string &local_frame_orientation, const std::string &body_frame_orientation,
-		Eigen::Affine3d &tf_parent2local, Eigen::Affine3d &tf_child2local,
-		Eigen::Affine3d &tf_parent2body, Eigen::Affine3d &tf_child2body)
+	void transform_lookup(const std::string &target, const std::string &source,
+				Eigen::Affine3d &tf_source2target)
 	{
 		try {
 			// transform lookup WRT local frame
-			tf_parent2local = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
-				frame_id, local_frame_orientation,
-				ros::Time(0)));
-			tf_child2local = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
-				child_frame_id, local_frame_orientation,
-				ros::Time(0)));
-
-			// transform lookup WRT body frame
-			tf_parent2body = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
-				frame_id, body_frame_orientation,
-				ros::Time(0)));
-			tf_child2body = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
-				child_frame_id, body_frame_orientation,
-				ros::Time(0)));
+			tf_source2target = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
+							target, source,
+							ros::Time(0)));
```

Comment:
Is this only ever used for static lookups? If not, then **correct timestamps** should be used, not latest.

Reply:
@kamilritz please fix this as well